source/agora/common/crypto/Schnorr.d

```diff
@@ -399,6 +399,69 @@ nothrow @nogc @safe unittest
     assert(stolen_key != kp.v);
 }
 
+/// Single-signer as part of multi-sig
+public Signature multiSigSign (const ref Pair R,
+    const ref Scalar v, const ref Scalar c) nothrow @nogc @safe
+{
+    Scalar s = R.v + (v * c);
+    return Sig(R.V, s).toBlob();
+}
+
+/// multi-sig verify
+public bool multiSigVerify (const ref Signature signature,
+    const ref Point sumV, const ref Scalar c) nothrow @nogc @safe
+{
+    Sig sigRs = Sig.fromBlob(signature);
+    return sigRs.s.toPoint() == sigRs.R + (sumV * c);
+}
+
+public Signature multiSigCombine (const ref Signature[] signatures) nothrow @nogc @safe
```

Comment:
Why not a range interface ? Also the `ref` seems superfluous.

Reply:
I updated to use templated range.